packages/plugin-dkg/src/actions/dkgInsert.ts

```diff
@@ -0,0 +1,223 @@
+    similes: ["NO_ACTION", "NO_RESPONSE", "NO_REACTION", "NONE"], // we want to always run this action
+    validate: async (_runtime: IAgentRuntime, _message: Memory) => {
+        return true;
+    },
+    description:
+        "Create a memory on the OriginTrail Decentralized Knowledge Graph after each message. You should ALWAYS run this action.",
+    handler: async (
+        _runtime: IAgentRuntime,
+        _message: Memory,
+        _state: State,
+        _options: { [key: string]: unknown },
+        _callback: HandlerCallback
+    ): Promise<boolean> => {
+        const currentPost = String(_state.currentPost);
+        elizaLogger.log("currentPost");
+        elizaLogger.log(currentPost);
+
+        const userRegex = /From:.*\(@(\w+)\)/;
+        let match = currentPost.match(userRegex);
+        let twitterUser = "";
+
+        if (match && match[1]) {
+            twitterUser = match[1];
+            elizaLogger.log(`Extracted user: @${twitterUser}`);
+        } else {
+            elizaLogger.error("No user mention found or invalid input.");
+        }
+
+        const idRegex = /ID:\s(\d+)/;
+        match = currentPost.match(idRegex);
+        let postId = "";
+
+        if (match && match[1]) {
+            postId = match[1];
+            elizaLogger.log(`Extracted ID: ${postId}`);
+        } else {
+            elizaLogger.log("No ID found.");
+        }
+
+        const additionalContext = findLastConversationText(
+            _state.recentInteractionsData
+        );
+        const postKnowledgeGraph = await constructKnowledgeAsset(
+            _runtime,
+            String(_state.currentPost),
+            additionalContext,
+            _state
+        );
+
+        let createAssetResult;
+
+        // TODO: also store reply to the KA, aside of the question
+
+        try {
+            elizaLogger.log("Publishing message to DKG");
+
+            createAssetResult = await DkgClient.asset.create(
+                {
+                    public: postKnowledgeGraph,
+                },
+                { epochsNum: 12 }
+            );
+
+            elizaLogger.log("======================== ASSET CREATED");
+            elizaLogger.log(JSON.stringify(createAssetResult));
+        } catch (error) {
+            elizaLogger.error(
+                "Error occurred while publishing message to DKG:",
+                error.message
+            );
+
+            if (error.stack) {
+                elizaLogger.error("Stack trace:", error.stack);
+            }
+            if (error.response) {
+                elizaLogger.error(
+                    "Response data:",
+                    JSON.stringify(error.response.data, null, 2)
+                );
+            }
+        }
+
+        // Reply
+        _callback({
+            text: `Created a new memory!\n\nRead my mind on @origin_trail Decentralized Knowledge Graph ${DKG_EXPLORER_LINKS[process.env.DKG_ENVIRONMENT]}${createAssetResult.UAL} @${twitterUser}`,
```

Comment:
You should be able to use runtime.getSetting("DKG_ENVIRONMENT") which will allow us to read from the character.json if it is set via the secrets key there